### PR TITLE
update ENA schemas to v1.16

### PR DESCRIPTION
**Pull request by blankdots**
### Description

include changes from:
- https://github.com/enasequence/schema/pull/29
- https://github.com/enasequence/schema/pull/27

### Type of change

- [x] Update schemas
### Changes Made

- update ENA xsd schemas with upstream, also `ena_analysis.json`
- analysis has the most changes (new file type) and new analysis types
- removed version numbers from xsd files (as done in ENA upstream files)
- refreshed wordlist
- added `assembly_graph_analysis.xml` to test the new schema additions
  - adjusted JSON parser to an interpretation mentioned in https://github.com/CSCfi/metadata-submitter/pull/628#issuecomment-1292345525

### Testing

- [x] Tests do not apply

### CODE REVIEW

**Review by csc-felipe — COMMENTED** [review]
Looks good. I don't understand these so much though, but looks like a tiny change, with few additions to the analysis schema.

Is the JSON schema addressing the changes enough?

**Comment by blankdots** [comment]
> Is the JSON schema addressing the changes enough?

In my biased expertise (biased, as i worked mostly on the ENA JSON schemas), i can say **yes**, the JSON schema is up to date. 
At the same time I can also say that in order to evaluate the interpretation of XSD to JSON schemas we need to test them with the users completing a form based on them, so most likely changes will required when we get feedback. Till such moment occurs all I can say, is that these are good enough for what our current understanding of the use cases is.

**Review by csc-jm — COMMENTED** [review]
Maybe we can take that [test file they included in the PR](https://github.com/enasequence/schema/pull/27/files#diff-35901316bef77dc6eac59c9707b8bc59082f8d91e3a8f30a22c2c151afce7d06) and include it in our tests (?)

**Comment by blankdots** [comment]
so it seems adjustments to the parser needed to be made so that the assembly graph test file passes, and the JSON structure would be something like:
```
{
    "alias": "graph_X_PGGB",
    "title": "Pangenome assembly graph X - generated using PGGB pipeline",
    "description": "Pangenome assembly graph X was generated \u2026. Etc. \u2026",
    "studyRef": {
      "accessionId": "ERP123456"
    },
    "sampleRef": [
      {
        "accessionId": "ERS123456"
      },
      {
        "accessionId": "ERS123457"
      },
      {
        "accessionId": "ERS123458"
      }
    ],
    "analysisType": {
      "assemblyGraph": [
        {
          "accession": "GCA_012345.1"
        },
        {
          "accession": "GCA_012346.1"
        },
        {
          "accession": "GCA_012347.1"
        }
      ]
    },
    "files": [
      {
        "filename": "assembly_graph.gfa.gz",
        "filetype": "gfa",
        "checksumMethod": "MD5",
        "checksum": "9f2976d079c10b111669b32590d1eb3e"
      }
    ],
    "analysisAttributes": [
      {
        "tag": "pangenome name",
        "value": "hprc-v1.0-mc-chm13-maxdel.10mb"
      },
      {
        "tag": "file format version",
        "value": "GFAv1.1"
      },
      {
        "tag": "pipeline",
        "value": "PGGB"
      },
      {
        "tag": "resolution",
        "value": "base level"
      },
      {
        "tag": "number of nodes",
        "value": "424643"
      },
      {
        "tag": "number of components",
        "value": "25"
      },
      {
        "tag": "number of edges",
        "value": "637628"
      },
      {
        "tag": "length (bp)",
        "value": "3239764787"
      },
      {
        "tag": "total path length (bp)",
        "value": "3239764787"
      },
      {
        "tag": "reference assembly",
        "value": "GRCh38"
      },
      {
        "tag": "reference assembly",
        "value": "ABCD123"
      },
      {
        "tag": "reference assembly",
        "value": "ABCD234"
      },
      {
        "tag": "reference based",
        "value": "true"
      }
    ]
  }
  ```
  for specific type of analysis it seems only one `assembly` is allowed based on XSD (more precisely for `"processedReads", "referenceAlignment", "sequenceAnnotation", "assemblyAnnotation"`) so this interpretation should be consistent for other analysis types.

File: metadata_backend/helpers/parser.py [hunk 1]
```diff
@@ -125,13 +125,31 @@ def _flatten(self, data: ElementData) -> Union[Dict, List, str, None]:
...
+                continue
+
+            if key == "assemblyGraph" and "assembly" in value:
+                attribs = value["assembly"]
+                attr_list = []
+                if isinstance(attribs, list):
+                    for d in attribs:
+                        for k, v in d.items():
+                            v["accession"] = v.pop("accessionId")
+                            attr_list.append(v)
```
Comment (csc-jm) [inline metadata_backend/helpers/parser.py:146]: Surprising that the style tests don't complain about this many nested statements :D

**Review by csc-jm — APPROVED** [review]
Seems good now

File: metadata_backend/helpers/parser.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/helpers/parser.py:146]: yeah, surprised as well, also noticed some comments there, will remove those ...

**Comment by blankdots** [comment]
@csc-jm you might need to re-approve as now when new updates are added, approves are dismissed

**Review by csc-jm — APPROVED** [review]

**Pull request merged**